scripts/create_deprecated_module.lean

```diff
@@ -0,0 +1,175 @@
+/-
+Copyright (c) 2025 Damiano Testa. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Damiano Testa
+-/
+
+--import Mathlib.Init
+import Lean.Elab.Command
+import Mathlib.Tactic.Linter.DeprecatedModule
+-- a comment here
+
+/-!
+#  Create a deprecated module
+
+This file defines the lean script for creating a deprecated module.
+-/
+
+open Lean Elab Command
+
+namespace DeprecatedModule
+
+/--
+`getHeader fname fileContent keepTrailing` takes as input two strings and a `Bool`ean.
+It uses
+* `fname`, as the path of a file (which need not exist);
+* `fileContent`, as the content of `fname` (regardless of whether the file exists and what its
+  content it);
+* `keepTrailing` a boolean.
+
+It returns the content of `fileContent` up to the final import, including trailing whitespace and
+comments if `keepTrailing = true`.
```

Comment:
If I'm reading the code right, comments are always kept - but whitespace only if keepTrailing is true. Is that correct - if so, can you reword the comment, please?

Reply:
Regardless of what `keepTrailing` is, the *actually trailing* whitespace is removed.

However, if `keepTrailing = true` then all trailing whitespace up to and including the last comment are kept, with the understanding that "no comment" means "no whitespace" as well!  Basically, with "whitespace" I was referring to the whitespace between the strict end of the syntax and end of the final comment.  Whitespace after the last comment is always discarded.

I am not sure that explaining this in such detail is more useful than leaving as is, though.  :smile:

Reply:
I like the solution you chose.